### DISCUSSION TITLE
External SQL Query - subpage

### DISCUSSION THREAD

**Discussion by ksl28** [Budibase/budibase#8237] [Help] [answered]
Hi,

I want to have budibase present my data, that is based on external SQL Queries.
Ive made an very simple SQL Database, that holds these 2 tables, with an PK on SQLInstances.dbo.SQLInstance and foreign key relationship from SQLInstanceStats.

![image](https://user-images.githubusercontent.com/53445047/183826629-80df4d47-3fe0-4158-8418-204986485a5e.png)

I managed to get the data provider to show me the data from the SQLInstances table, in an nice table on the "front page".
![image](https://user-images.githubusercontent.com/53445047/183827018-5b924736-8234-4fba-8397-75a3870c5e6e.png)

The goal is that when you press the View button, you are redirected to an subpage, that then executes an SQL query against SQLInstanceStats - fx 
`
SELECT * FROM SQLInstanceStats  
WHERE SQLInstance = 'SQLInstanceFromBudiBase'
`

That way i get all the data from my SQL Database, but only for the instance i clicked.
But i cant seem to figure out, how to send the SQLInstance value to the query.

Any ideas how to accomplish this?

**Comment by melohagan** [comment]
Hey @ksl28 

Sorry for the late response, I missed it because it was initially an issue and not a discussion. 

You can pass binding values through to custom queries as outlined here: https://docs.budibase.com/docs/custom-queries

**Comment by ksl28** [comment, marked as answer]
Hey @melohagan ,

Thanks for the update, really appreciate it! :) 
Your guide makes perfect sense, but im still stuck on how to use the unique value in the query.

![image](https://user-images.githubusercontent.com/53445047/200915319-757408cd-1a51-4b0c-b1c6-211ca87e4ad5.png)


Since the title uses the Repeater.SQLInstances.SQLInstance, i thought i might be able to use it in the query as well - but without any luck.
If i enter something static (SQLSRV1), then it pulls the data correctly, but it uses SQLSRV1 for all the queries.

What is the correct value, to enter in there?

**Reply by melohagan** [reply]
`Repeater.SQLInstances` looks like an array? Also you're not using the handlebars. I think what you have highlighted in red should probably be something like: `{{ Repeater.SQLInstances.0.SQLInstance }}`

If you click the bolt icon you should see a list of available bindings that you can add by clicking on them.

**Reply by ksl28** [reply]
Hi,
Thanks for replying :) 
It turned out i had the data provider outside my repeater, so it did not work - after i moved it inside, i was able to make it work :)

![image](https://user-images.githubusercontent.com/53445047/201040737-1c6c1837-87d1-49a0-9d84-e100b2522104.png)

![image](https://user-images.githubusercontent.com/53445047/201040841-4b2a2baf-f7cf-4a73-89cc-a7609a2fab5a.png)


Budibase is really an amazing tool - but it takes some time to learn :)

**Reply by melohagan** [reply]
Nice one, good job!
